extensions/wikia/ImageReview/ImageReviewHelper.class.php

```diff
@@ -203,221 +235,283 @@ public function refetchImageListByTimestamp( $timestamp ) {
 	}
 
 	/**
-	 * get image list
+	 * Get image list
+	 *
+	 * @param $timestamp
+	 * @param int $state
+	 * @param int $order
+	 * 
 	 * @return array imageList
+	 *
+	 * @throws DBUnexpectedError
+	 * @throws Exception
+	 * @throws MWException
 	 */
 	public function getImageList( $timestamp, $state = ImageReviewStatuses::STATE_UNREVIEWED, $order = self::ORDER_LATEST ) {
```

Comment:
This method was loading 128 images, changing their state to be in review for the current user, then picking 20 out of that list and changing the remaining 108 images (minus any icons or bad images) back to un-reviewed.  While this was going on, there was nothing stopping another user form getting the same batch of images and updating them.

Rather than do all this busy work, immediately assign each image one by one with an atomic update, decide if its worth keeping, send it to another queue (delete or icon review) if its not.  This is all done by an iterator that will keep fetching images so we don't have to get an arbitrary number and hope we're covered.  I kept the 128 limit in however as a hard limit however, but that could be increased to something much higher.